docs/source/miscellaneous/doc/adr/0014-cassandra-targeted-sweep.md

```diff
@@ -0,0 +1,283 @@
+  in, or range scanning the much denser *sweepableTimestamps* table to locate the next coarse partition with entries and then reading
+  the fine partition to look for.
+
+#### sweepProgressPerShard
+
+  This table stores targeted sweep's progress, as well as the information about the number of shards the sweep queue is using.
+
+  **Row components**:
+  - **shard**: a `VAR_SIGNED_LONG` containing the shard for which we are tracking progress for.
+  - **sweep_conservative**: a `boolean` (encoded as a `BLOB`) specifying if we are looking for conservative or thorough sweep as in
+    *sweepableTimestamps*.
+
+  **Named Column**:
+  - **value**: a `VAR_LONG` containing the timestamp up to which targeted sweep has swept on that shard and strategy.
+
+  To persist the number of shards sweep queue is using, we use a distinguished row of this table, with the row defined by
+  *shard = -1* and *sweep_conservative = true*. Note that all writes to this table use atomic check and set, and the values are only
+  allowed to be increased. A request to update a cell to a lower value will have no effect.
+
+#### Writing to the Sweep Queue
+
+Whenever a transaction is about to commit, before its writes are persisted into the KVS, `SerializableTransactionManager` enqueues
+them to the sweep queue. Sweep queue creates a list of `WriteInfo`s containing the relevant information, partitions this list
+according to the sweep strategies for the tables (this information is read from the table metadata and then cached), and into the
+number of shards that the sweep queue is using (the shard is determined from the hash of the `TableReference` and `Cell`). For each
+of these partitions, we then put an entry into the *sweepableTimestamps* table for the start timestamp of the transaction. Finally,
+for each of the partitions, we put the `WriteInfo`s into the *sweepableCells* table:
+  - If there are 50 or fewer entries in the list, write that many cells into the table where all the row and column components are
+    calculated as described above, with the *write_index* starting at 0 and increasing by 1 for each entry.
+  - If there are more than 50 entries in the list, put a single cell into the table acting as a reference to dedicated rows. The row
+    of the reference is the same as above, but the write_index is a negative number with an absolute value equal to the number of
+    dedicated rows we will use (number of writes divided by 100_000, rounded up). Then, put the cells into dedicated rows, where
+    the *timestamp_partition* is the start timestamp of the transaction (**not the fine partition!**), the metadata encodes that the
+    row is a dedicated row and the row's ordinal number, and the rest is the same as above.
+
+#### Reading from the Sweep Queue
+
+Reading from the sweep queue is done in the reverse order from writing. For a given shard, strategy, **minimum exclusive
```

Comment:
It seems to me that we read in the same order things are written - in increasing timestamp order?

Reply:
Ah, I was trying to say something else here, but it ended up being just messed up